roda-core/roda-core/src/main/java/org/roda/core/RodaCoreFactory.java

```diff
@@ -1515,17 +1515,21 @@ private static void instantiateNodeSpecificObjects(NodeType nodeType) {
     }
   }
 
+  private static void instantiateAccessTokens(){
+    apiSecretKey = getProperty(RodaConstants.API_SECRET_KEY_PROPERTY, RodaConstants.DEFAULT_API_SECRET_KEY);
+    accessKeyValidity = RodaCoreFactory.getRodaConfiguration().getLong(RodaConstants.ACCESS_KEY_VALIDITY,
+      RodaConstants.DEFAULT_ACCESS_KEY_VALIDITY);
+    accessTokenValidity = RodaCoreFactory.getRodaConfiguration().getLong(RodaConstants.ACCESS_TOKEN_VALIDITY,
+      RodaConstants.DEFAULT_ACCESS_TOKEN_VALIDITY);
+  }
+
   private static void instantiateDistributedMode() {
     initializeLocalInstanceConfigDirectory();
     distributedModeType = DistributedModeType.valueOf(
       getProperty(RodaConstants.DISTRIBUTED_MODE_TYPE_PROPERTY, RodaConstants.DEFAULT_DISTRIBUTED_MODE_TYPE.name()));
     try {
+      instantiateAccessTokens();
```

Comment:
Move the call to the same level instantiateDistributedMode is called, not called inside instantiateDistributedMode.

For example between those two calls
https://github.com/keeps/roda/blob/8b9675aa8d0b151fa0b8701ed3e96b69022085d8/roda-core/roda-core/src/main/java/org/roda/core/RodaCoreFactory.java#L528-L532

Add a finish log information

Reply:
done